controllers/provider-azure/pkg/internal/infrastructure/terraform.go

```diff
@@ -134,6 +141,14 @@ func ComputeTerraformerChartValues(infra *extensionsv1alpha1.Infrastructure, cli
 		azure["countFaultDomains"] = countFaultDomains
 	}
 
+	if config.Identity != nil && config.Identity.Name != "" && config.Identity.ResourceGroup != "" {
```

Comment:
See above - I would only check for `Identity != nil` here and handle the potential empty `Name/ResourceGroup` via validation.